localstack/services/sns/publisher.py

```diff
@@ -864,25 +864,87 @@ def create_unsubscribe_url(external_url, subscription_arn):
-            attribute = message_attributes.get(criteria)
+        return True
 
-            if not self._evaluate_filter_policy_conditions(
-                conditions, attribute, message_attributes, criteria
-            ):
+    def check_filter_policy_on_message_body(self, filter_policy: dict, message_body: str):
+        try:
+            body = json.loads(message_body)
+            if not isinstance(body, dict):
                 return False
+        except json.JSONDecodeError:
+            # Filter policies for the message body assume that the message payload is a well-formed JSON object.
+            # See https://docs.aws.amazon.com/sns/latest/dg/sns-message-filtering.html
+            return False
+
+        return self._evaluate_nested_filter_policy_on_dict(filter_policy, payload=body)
+
+    def _evaluate_nested_filter_policy_on_dict(self, filter_policy, payload: dict) -> bool:
+        """
+        This method evaluate the filter policy recursively, while still being able to validate the `exists` condition.
+        Example:
+        nested_filter_policy = {
+            "object": {
+                "key": [{"prefix": "auto-"}],
+                "nested_key": [{"exists": False}],
+            },
+            "test": [{"exists": False}],
+        }
+        payload = {
+            "object": {
+                "key": "auto-test",
+            }
+        }
+        This function then iterates on top level keys of the filter policy: ("object", "test")
+        The value of "object" is a dict, we need to evaluate this level of the filter policy.
+        We pass the nested property values (the dict) as well as the values of the payload's field to the recursive
+        function, to evaluate the conditions on the same level of depth.
+        We then these parameters:
+        filter_policy = {
+            "key": [{"prefix": "auto-"}],
+            "nested_key": [{"exists": False}],
+        }
+        payload = {
+            "key": "auto-test",
+        }
+        We can now properly evaluate the conditions on the same level of depth in the dict object.
+        As it passes the filter policy, we then continue to evaluate the top keys, going back to "test".
+        :param filter_policy: a dict, starting at the FilterPolicy
+        :param payload: a dict, starting at the MessageBody
+        :return: True if the payload respect the filter policy, otherwise False
+        """
+        for field_name, values in filter_policy.items():
+            # if values is not a dict, then it's a nested property
+            if not isinstance(values, list):
+                if not self._evaluate_nested_filter_policy_on_dict(
+                    values, payload.get(field_name, {})
+                ):
+                    return False
+            else:
+                # else, values represents the list of conditions of the filter policy
+                for condition in values:
+                    print(condition)
```

Comment:
Is this print intended? If yes, maybe add something like `print(f"Filter policy condition: {condition}")`

Reply:
Oops, that shouldn't be here, I'll remove! thanks!